android/lib/src/main/java/com/swmansion/gesturehandler/PanGestureHandler.java

```diff
@@ -221,7 +224,7 @@ protected void onHandle(MotionEvent event) {
     if (action == MotionEvent.ACTION_UP) {
       if (state == STATE_ACTIVE) {
         end();
-      } else {
+      } else if (state == STATE_BEGAN) {
```

Comment:
Are you sure that this handler is going to be removed from active handlers list after getting UP event if it was in UNDETERMINED state before?